File: StkExtensionPlugins/RT3/CSharp/AGI.Realtime.Examples.Providers/ProvideEntitiesFromTextFileForm.cs
using System;
using System.Runtime.InteropServices;
using System.Windows.Forms;

namespace AGI.Realtime.Examples.Providers
{
    /// <summary>
    /// A dialog plug-in which configures an instace of ProvideEntitiesFromFile.
    /// 
    /// This class implements IAgUiRtDialog, which allows for an application to
    /// discover a GUI at runtime which can be used to configure the associated
    /// plug-in.  It is the only required interface for a dialog plug-in.
    /// 
    /// To use this class in RT3, simply compile the library and load the
    /// included reg file into the Windows registry.
    /// 
    /// For more information on RT3 interfaces and class, please see the RT3
    /// Development Kit documentation.
    /// </summary>
    [Guid(ProvideEntitiesFromTextFileForm.ClassId),
    ProgId(ProvideEntitiesFromTextFileForm.ProgID),
    ClassInterface(ClassInterfaceType.None)]
    public partial class ProvideEntitiesFromTextFileForm :
        Form,
        IAgUiRtDialog
    {
        #region ProvideEntitiesFromTextFileForm Members

        public ProvideEntitiesFromTextFileForm()
        {
            InitializeComponent();
        }

        private void ProvideEntitiesFromTextFileForm_Load(object sender, EventArgs e)
        {
            m_filename.Text = m_plugin.Filename;

            //If we are running, make the GUI read-only
            if (m_plugin.Active)
            {
                m_browse.Enabled = false;
            }
        }

        private void m_browse_Click(object sender, EventArgs e)
        {
            OpenFileDialog dlg = new OpenFileDialog();
            dlg.FileName = m_filename.Text;
            dlg.Filter = "All Files (*.*)|*.*";
            if (dlg.ShowDialog() == DialogResult.OK)
            {
                m_filename.Text = dlg.FileName;
            }
        }

        private void m_ok_Click(object sender, EventArgs e)
        {
            m_plugin.Filename = m_filename.Text;
            DialogResult = DialogResult.OK;
        }

        #endregion

        #region IAgUiRtDialog Members

        /// <summary>
        /// Gets or sets the plug-in to be configured.
        /// </summary>
        public object Data
        {
            get
            {
                return m_plugin;
            }
            set
            {
                m_plugin = value as ProvideEntitiesFromTextFile;
            }
        }

        /// <summary>
        /// Displays the dialog, either modal or modeless, and configures
        /// the plug-in that has been passed in as the Data property.
        /// </summary>
        /// <param name="Parent">The HWND of the parent window.</param>
        /// <returns>The result of the dialog.</returns>
        public AgEUiRtDialogResult Activate(ref int Parent)
        {
            if (ShowDialog() == DialogResult.Cancel)
            {
                return AgEUiRtDialogResult.eUiRtDialogResultCancel;
            }

            return AgEUiRtDialogResult.eUiRtDialogResultOK;
        }

        #endregion

        #region Variables, Constants, Events & Delegates
        
        //Every RT3 plug-in needs to have a unique ProgID.  You should change
        //the below ProgID to match any new classes you create.
        public const string ProgID = "AGI.Realtime.Examples.Providers.ProvideEntitiesFromTextFileForm";

        //Each RT3 plug-in needs to have it's own GUID.  Generate a new GUID
        //your class using GUIDGen, located in "Tools->Create GUID" in Visual
        //Studio or http://www.guidgen.com/
        public const string ClassId = "22232A1B-8C0C-4d25-BC3E-D30F11BB1453";
        
        private ProvideEntitiesFromTextFile m_plugin;

        #endregion
    }
}

File: StkExtensionPlugins/RT3/VB.NET/AGI.Realtime.Examples.Providers/ProvideEntitiesFromTextFileForm.vb
Imports System.Runtime.InteropServices
Imports System.Windows.Forms

''' <summary>
''' A dialog plug-in which configures an instace of ProvideEntitiesFromFile.
''' 
''' This class implements IAgUiRtDialog, which allows for an application to
''' discover a GUI at runtime which can be used to configure the associated
''' plug-in.  It is the only required interface for a dialog plug-in.
''' 
''' To use this class in RT3, simply compile the library and load the
''' included reg file into the Windows registry.
''' 
''' For more information on RT3 interfaces and class, please see the RT3
''' Development Kit documentation.
''' </summary>
<ComClass(ProvideEntitiesFromTextFileForm.ClassId, _
          ProvideEntitiesFromTextFileForm.InterfaceId)> _
Public Class ProvideEntitiesFromTextFileForm
    Implements IAgUiRtDialog
#Region "ProvideEntitiesFromTextFileForm Members"
    Private Sub ProvideEntitiesFromTextFileForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        m_filename.Text = m_plugin.Filename

        'If we are running, make the GUI read-only
        If m_plugin.Active Then
            m_browse.Enabled = False
        End If
    End Sub

    Private Sub m_browse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles m_browse.Click
        Dim dlg As New OpenFileDialog
        dlg.FileName = m_filename.Text
        If dlg.ShowDialog() = DialogResult.OK Then
            m_filename.Text = dlg.FileName
        End If
    End Sub

    Private Sub m_ok_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles m_ok.Click
        m_plugin.Filename = m_filename.Text
        DialogResult = DialogResult.OK
    End Sub
#End Region

#Region "IAgUiRtDialog Members"
    ''' <summary>
    ''' Gets or sets the plug-in to be configured.
    ''' </summary>
    Public Property Data() As Object Implements IAgUiRtDialog.Data
        Get
            Data = m_plugin
        End Get
        Set(ByVal value As Object)
            m_plugin = CType(value, ProvideEntitiesFromTextFile)
        End Set
    End Property

    ''' <summary>
    ''' Displays the dialog, either modal or modeless, and configures
    ''' the plug-in that has been passed in as the Data property.
    ''' </summary>
    ''' <param name="Parent">The HWND of the parent window.</param>
    ''' <returns>The result of the dialog.</returns>
    Public Function IAgUiRtDialog_Activate(ByRef Parent As Integer) As AgEUiRtDialogResult Implements IAgUiRtDialog.Activate
        If ShowDialog() = DialogResult.Cancel Then
            IAgUiRtDialog_Activate = AgEUiRtDialogResult.eUiRtDialogResultCancel
        Else
            IAgUiRtDialog_Activate = AgEUiRtDialogResult.eUiRtDialogResultOK
        End If
    End Function

    Public ReadOnly Property IAgUiRtDialog_Name() As String Implements IAgUiRtDialog.Name
        Get
            IAgUiRtDialog_Name = "ProvideEntitiesFromTextFileForm"
        End Get
    End Property
#End Region

#Region "Variables, Constants, Events & Delegates"
    'Each RT3 plug-in needs to have it's own GUID.  Generate a new GUID
    'your class using GUIDGen, located in "Tools->Create GUID" in Visual
    'Studio or http://www.guidgen.com/
    Public Const ClassId As String = "5F941829-D5C2-4f45-AC13-D1F47B81B823"
    Public Const InterfaceId As String = "CA8E9600-4877-4dea-95AA-669DD9FBC9FD"
    Private m_plugin As ProvideEntitiesFromTextFile
#End Region
End Class